lib/src/rules/prefer_contains.dart

```diff
@@ -0,0 +1,219 @@
+```
+if (lunchBox.length == 0) return 'so hungry...';
+if (words.length != 0) return words.join(' ');
+```
+''';
+
+class _LintCode extends LintCode {
+  static final registry = <String, LintCode>{};
+
+  factory _LintCode(String name, String message) => registry.putIfAbsent(
+      name + message, () => new _LintCode._(name, message));
+
+  _LintCode._(String name, String message) : super(name, message);
+}
+
+class PreferContains extends LintRule {
+  PreferContains()
+      : super(
+            name: 'prefer_contains',
+            description: desc,
+            details: details,
+            group: Group.style);
+
+  @override
+  AstVisitor getVisitor() => new Visitor(this);
+
+  void reportLintWithDescription(AstNode node, String description) {
+    if (node != null) {
+      reporter.reportErrorForNode(new _LintCode(name, description), node, []);
+    }
+  }
+}
+
+class Visitor extends SimpleAstVisitor {
+  final PreferContains rule;
+
+  Visitor(this.rule);
+
+  @override
+  visitSimpleIdentifier(SimpleIdentifier identifier) {
+    // Should be "indexOf".
+    Element propertyElement = identifier.bestElement;
+    if (propertyElement?.name != 'indexOf') {
+      return;
+    }
+
+    AstNode lengthAccess;
+    InterfaceType type;
+
+    AstNode parent = identifier.parent;
+    if (parent is MethodInvocation && identifier == parent.methodName) {
+      lengthAccess = parent;
+      if (parent.target?.bestType is! InterfaceType) {
+        return;
+      }
+      type = parent.target?.bestType;
+    } else {
+      return;
+    }
+
+    // Should be subtype of Iterable or String.
+    AnalysisContext context = propertyElement.context;
+    TypeProvider typeProvider = context.typeProvider;
+    TypeSystem typeSystem = context.typeSystem;
+
+    if (typeSystem.mostSpecificTypeArgument(type, typeProvider.iterableType) ==
+            null &&
+        !type.element.type.isSubtypeOf(typeProvider.stringType)) {
+      return;
+    }
+
+    AstNode search = lengthAccess;
+    while (
+        search != null && search is Expression && search is! BinaryExpression) {
+      search = search.parent;
+    }
+
+    if (search is! BinaryExpression) {
+      return;
+    }
+    BinaryExpression binaryExpression = search;
+
+    Token operator = binaryExpression.operator;
+    DeclaredVariables declaredVariables = context.declaredVariables;
+
+    // Comparing constants with result of indexOf.
+
+    ConstantVisitor visitor = new ConstantVisitor(
+        new ConstantEvaluationEngine(typeProvider, declaredVariables,
+            typeSystem: typeSystem),
+        new ErrorReporter(
+            AnalysisErrorListener.NULL_LISTENER, rule.reporter.source));
+
+    DartObjectImpl rightValue = binaryExpression.rightOperand.accept(visitor);
+
+    if (rightValue?.type?.name == "int") {
+      // Constant is on right side of comparison operator
+      int value = rightValue.toIntValue();
+      if (value == -1) {
+        if (operator.type == TokenType.EQ_EQ ||
```

Comment:
I'm not sure that it will be faster or easier to understand. All conditions are different. But I have an idea how to reduce code size, maybe it will help.